web/src/components/planner/products/BuildingGroup.vue

```diff
@@ -0,0 +1,330 @@
+      <div>=</div>
+      <div class="underchip">&nbsp;</div>
+    </div>
+    <template v-for="(_, part) in group.parts" :key="`${item.id}-${part}`">
+      <div v-if="part.toString() === item.id || partIsByProduct(String(part), group.type)">
+        <v-chip
+          class="sf-chip input mx-1 text-body-1"
+          :class="chipColors(String(part))"
+          variant="tonal"
+        >
+          <tooltip :text="getPartDisplayName(part)">
+            <game-asset :subject="String(part)" type="item" />
+          </tooltip>
+          <v-number-input
+            v-model.number="group.parts[part]"
+            class="inline-inputs"
+            control-variant="stacked"
+            density="compact"
+            hide-details
+            hide-spin-buttons
+            :min="0"
+            :name="`${item.id}-${part}.amount`"
+            width="110px"
+          />
+        </v-chip>
+        <div class="underchip" :class="partIsByProduct(String(part), group.type) ? 'text-grey-lighten-2' : 'text-blue-darken-1'">
+          {{ formatNumberFully(group.parts[part] / group.buildingCount) }} / building
+        </div>
+      </div>
+    </template>
+  </div>
+</template>
+
+<script setup lang="ts">
+  import { defineProps, inject } from 'vue'
+  import {
+    BuildingGroup,
+    Factory,
+    FactoryItem,
+    FactoryPowerProducer,
+    GroupType,
+  } from '@/interfaces/planner/FactoryInterface'
+  import eventBus from '@/utils/eventBus'
+  import { getPartDisplayName } from '@/utils/helpers'
+  import { useGameDataStore } from '@/stores/game-data-store'
+  import { increaseProductQtyViaBuilding } from '@/utils/factory-management/products'
+  import { useDisplay } from 'vuetify'
+  // import { updateGroupParts } from '@/utils/factory-management/productBuildingGroups'
+  import { formatNumberFully, formatPower } from '@/utils/numberFormatter'
+  import { PowerItem } from '@/interfaces/Recipes'
+
+  const updateFactory = inject('updateFactory') as (factory: Factory) => void
+  const gameData = useGameDataStore().getGameData()
+
+  // const timeout: NodeJS.Timeout | null = null
+  let updatingPart: string
+
+  const { lgAndDown, lgAndUp } = useDisplay()
+
+  const props = defineProps<{
+    factory: Factory
+    group: BuildingGroup
+    item: FactoryItem | FactoryPowerProducer
+    building: string // Building name
+  }>()
+
+  const updateGroup = (group: BuildingGroup) => {
+    if (group.buildingCount === 0 || isNaN(group.buildingCount) || group.buildingCount === null) {
+      eventBus.emit('toast', {
+        message: 'Building count must be a positive number.',
+        type: 'warning',
+      })
+      group.buildingCount = 1
+      return
+    }
+
+    // Ensure the building count is a whole number
+    if (group.buildingCount % 1 !== 0) {
+      eventBus.emit('toast', {
+        message: 'Building count must equal to a whole number. If you need a single building clocked, create a new building group and adjust it\'s clock.',
```

Comment:
definitely a nitpick but you want "its" here, not "it's" (it's is "it is" or "it has")